### PR TITLE
Order users on delegation page, selected delegate first.

### LINKED ISSUE #817: show phase in idea detail view (like in topic detail view)

### LINKED ISSUE #821: user order on "delegate on topic / idea level" page.
currently, the delegate is hidden in the hundrets of users.  she should be on top of the list.

### PR DESCRIPTION
**Pull request by fisx**
Fixes #821.  Fixes #817.

### CODE REVIEW

File: src/Frontend/Page/Idea.hs [hunk 1]
```diff
@@ -91,7 +93,12 @@ import qualified Lucid
 -- * 5.4 Idea detail page: Voting phase
 -- * 5.6 Idea detail page: Feasible / not feasible
 -- * 5.7 Idea detail page: Winner
-data ViewIdea = ViewIdea { _viCtx :: CapCtx, _viStats :: IdeaStats, _viDelegation :: Maybe Delegation }
+data ViewIdea = ViewIdea
+    { viNow :: Timestamp
```
Comment (andorp) [inline src/Frontend/Page/Idea.hs]: Please make it a lens to be consistent.

File: src/Action.hs [hunk 2]
```diff
@@ -371,15 +371,9 @@ validUserState us = us == userLoggedOut || validLoggedIn us
 
 getSpacesForCurrentUser :: (ActionUserHandler m, ActionPersist m) => m [IdeaSpace]
 getSpacesForCurrentUser = do
-    -- FIXME: remove the isLoggedIn check.
```
Comment (andorp) [inline src/Action.hs]: Why we had this FIXME here? Where is the implication of the removal of the `if`?

File: src/Frontend/Page/Delegation.hs [hunk 3]
```diff
@@ -77,7 +78,16 @@ instance FormPage PageDelegateVote where
...
-    formPage v f p@(PageDelegateVote scope options _mselected) = semanticDiv p . f $ do
+    formPage v f p@(PageDelegateVote scope options mselected) = semanticDiv p . f $ do
+        let options' = sortBy (maybe compare compareOptions mselected) options
+
+            -- always move selected user to the top.
+            compareOptions :: AUID User -> User -> User -> Ordering
+            compareOptions uid u1 u2
+                | u1 ^. _Id == uid = LT
+                | u2 ^. _Id == uid = GT
+                | otherwise        = compare (u1 ^. userLogin) (u2 ^. userLogin)
```
Comment (andorp) [inline src/Frontend/Page/Delegation.hs]: :+1:

File: src/Action.hs (hunk 2 above)
Comment (fisx) [inline src/Action.hs]: see commit message.

File: src/Frontend/Page/Idea.hs (hunk 1 above)
Comment (fisx) [inline src/Frontend/Page/Idea.hs]: can't -- that would make -Wall get grumpy.

File: src/Action.hs (hunk 2 above)
Comment (andorp) [inline src/Action.hs]: Explanation accepted.

File: src/Frontend/Page/Idea.hs (hunk 1 above)
Comment (fisx) [inline src/Frontend/Page/Idea.hs]: ok, explicit export.  didn't think of that, now i did.  (-:

**Comment by andorp** [comment]
LGTM

**Pull request merged**